packages/composites/anchor/src/endpoint/price/beth.ts

```diff
@@ -0,0 +1,38 @@
+import { BigNumber, ethers } from 'ethers'
+import { PriceExecute } from '.'
+
+export const FROM = 'BETH'
+export const INTERMEDIARY_TOKEN_DECIMALS = 8
+export const INTERMEDIARY_TOKEN = 'stETH'
+
+export const anchorVaultAbi = [
+  {
+    stateMutability: 'view',
+    type: 'function',
+    name: 'get_rate',
+    inputs: [],
+    outputs: [{ name: '', type: 'uint256' }],
+  },
+]
+
+export const execute: PriceExecute = async (input, _, config, taAdapterResponse) => {
+  const rpcUrl = config.rpcUrl
+  const provider = new ethers.providers.JsonRpcProvider(rpcUrl)
+  const anchorVaultContract = new ethers.Contract(
+    config.anchorVaultContractAddress,
+    anchorVaultAbi,
+    provider,
+  )
+  const bEthExchangeRateBigNum = await anchorVaultContract.get_rate()
```

Comment:
Could we just use `terra-view-function` adapter for this?

Reply:
we can't as the `anchorVaultContract` isn't on Terra

Reply:
Ah I gotcha. Yeah that would be good for the "more general" implementation in the future.